### DISCUSSION TITLE
Lightning Strike Data

### DISCUSSION THREAD

**Discussion by jazzyisj** [Pirate-Weather/pirateweather#212] [Ideas]
### Describe the feature

Any chance Blitzortung data could be incorporated for real-time lightning strike sensors?   There must be an API.  There is already an HA integration that does, but it doesn't seem to be maintained regularly.   I did search the discussions and issues, I'm surprised this hasn't been suggested already.

https://github.com/mrk-its/homeassistant-blitzortung

https://www.blitzortung.org/en/live_lightning_maps.php

### Acknowledgements

- [X] I have searched this repository and [Home Assistant Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the feature has already been requested.
- [X] I have written an informative title.

**Comment by ktrue** [comment]
Alas, the Blitzortung.org network is a set of hobbyist constructed/operated lightning sensors and has some strict rules about reuse of the data provided.
1) to get the data, you have to contribute data from your personal sensor(s) to the network.
2) access to the data is based on Blitzortung user:pass
3) there is no API for public access to the direct data.

I offer a [script set](https://github.com/ktrue/BO-mapgen) to generate animated GIF strike maps, but.. you have to be a contributing member to Blitzortung to use it. 
I've been running a Blitzortung RED sensor I built since October, 2013.

**Comment by jazzyisj** [comment]
Thank you for the info @ktrue. That explains why I couldn't find anything about an API on their website.  I wonder how HA integration is pulling data?  

I know Environment Canada has a lightning map but I don't think they publish the data as far as I can tell.

https://weather.gc.ca/lightning/index_e.html?id=NAT#mapTop

**Comment by ktrue** [comment]
If you dig into his python code, there's a part where the Blitzortung user:pass is inserted in the call to get data.
I assume the author has a Blitzortung detector submitting data so they have Blitzortung credentials to use.

AFAIK, there's no public (free) lightning strike data available except images of strikes from NOAA satellites (but nothing but images).
The Vaisala Precision Lightning Network is a pay-for item (offered via api.xweather.com).  The Weather Channel API also has lightning data, also pay-for (and likely uses a resale of the Vaisala data)

**Comment by jazzyisj** [comment]
Thanks for considering it.  I'm going to try and touch base with Environment Canada to see if they are any plans to add this data to their public API's and if not maybe contact my MP to encourage it.  Probably won't get anywhere but it's worth a shot.

**Reply by cloneofghosts** [reply]
I've moved this over to the discussion board so you can reply here if you ever find a source of data or Blitzortung releases an API.

**Reply by jazzyisj** [reply]
Awesome!  BTW  I meant to include the envcan map.  This is the data I hope they might be able to put into their weather API.

https://weather.gc.ca/lightning/index_e.html
